AlarmPlus/Views/AlarmSettingsViewController.swift

```diff
@@ -0,0 +1,182 @@
+            
+            buttonPressed.isSelected = true
+            if alarmCurrentDays.contains(buttonPressed.tag) == false {
+                alarmCurrentDays.append(buttonPressed.tag)
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        
+        if alarm == nil {
+            setupNewAlarmInfo()
+        } else {
+            setupAlarmInfo()
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        
+    }
+    
+    
+    func setupView() {
+        let saveButton = UIBarButtonItem.init(title: "Save", style: .plain, target: self, action: #selector(rightButtonAction(sender:)))
+        navigationItem.rightBarButtonItem = saveButton
+        daySelectButtonsArray = [sundayButton, mondayButton, tuesdayButton, wednesdayButton, thursdayButton, fridayButton, saturdayButton, sundayButton]
+    }
+    
+    @objc func rightButtonAction(sender: UIBarButtonItem) {
+        //Save UIBarbutton item actions
+        var alarmsArray = MemoryFunctions.getSavedAlarmsArray()
+        
+        guard let currentAlarm = alarm else {
+            return
+        }
+        
+        let (setHour, setMinute) = getUserSetTime(datePicker: datePicker)
+        
+        currentAlarm.schedule.setAlarmTime(hour: setHour, minute: setMinute)
+
+        
+        currentAlarm.schedule.setDayaActive(daysArray: alarmCurrentDays)
+
+        
+        currentAlarm.schedule.setActiveAlarms()
+        
+        alarmsArray.append(currentAlarm)
+
+//        let data = MemoryFunctions.archiveWithKeyedArchiver(object: alarmsArray)
+//
+//        do {
+//            print(try MemoryFunctions.unarchiveAlarmArrayData(data))
+//        } catch {
+//            print(error.localizedDescription)
+//        }
+        
+        MemoryFunctions.saveAlarmsToUserDefaults(alarmsArray)
+
+    }
+    
+    //Function gets the information from existing alarm and pupulates it on the view
+    func setupAlarmInfo() {
+        
+    }
+    
+    //Function creates a new alarm object and sets it with some default data
+    func setupNewAlarmInfo() {
+        alarm = Alarm(snooze: Snooze(), alert: Alert(), schedule: Schedule())
+        
+        let calendar = Calendar(identifier: .gregorian)
+        let date = Date()
+        var dateCompenents = calendar.dateComponents(in: .current, from: date)
+        
+        dateCompenents.hour = 12
+        dateCompenents.minute = 0
+        
+        if let todaysDayOfTheWeekIndex = dateCompenents.weekday {
+            alarmCurrentDays.append(todaysDayOfTheWeekIndex)
+            
+            for dayButton in daySelectButtonsArray {
+                if dayButton.tag == todaysDayOfTheWeekIndex {
+                    dayButton.isSelected = true
+                }
+            }
+        }
+        datePicker.date = calendar.date(from: dateCompenents)!
+    }
+    
+    
+    func getUserSetTime(datePicker: UIDatePicker) -> (Int, Int){
```

Comment:
Usually returned variables have a name. So for this line, you could do:
`func getUserSetTime(datePicker: UIDatePicker) -> (hour: Int, minute: Int) {`